sormas-ui/src/main/java/de/symeda/sormas/ui/samples/sampleLink/SampleListComponent.java

```diff
@@ -45,33 +38,7 @@
 @SuppressWarnings("serial")
 public class SampleListComponent extends VerticalLayout {
 
-	private SampleList list;
-
-	public SampleListComponent(ContactReferenceDto contactRef, Disease disease, AbstractDetailView<? extends ReferenceDto> view) {
-		createSampleListComponent(
-			new SampleList(new SampleListCriteria.Builder().withContact(contactRef).build()),
-			e -> ControllerProvider.getSampleController().create(contactRef, disease, SormasUI::refreshView),
-			view);
-	}
-
-	public SampleListComponent(CaseReferenceDto caseRef, Disease disease, AbstractDetailView<? extends ReferenceDto> view) {
-		createSampleListComponent(
-			new SampleList(new SampleListCriteria.Builder().withCase(caseRef).build()),
-			e -> ControllerProvider.getSampleController().create(caseRef, disease, SormasUI::refreshView),
-			view);
-	}
-
-	public SampleListComponent(EventParticipantReferenceDto eventParticipantRef, Disease disease, AbstractDetailView<? extends ReferenceDto> view) {
-		createSampleListComponent(
-			new SampleList(new SampleListCriteria.Builder().withEventParticipant(eventParticipantRef).build()),
-			e -> ControllerProvider.getSampleController().create(eventParticipantRef, disease, SormasUI::refreshView),
-			view);
-	}
-
-	private void createSampleListComponent(
-		SampleList sampleList,
-		Consumer<Button.ClickEvent> clickListener,
-		AbstractDetailView<? extends ReferenceDto> view) {
+	public SampleListComponent(SampleList sampleList, Consumer<Button.ClickEvent> clickListener, AbstractDetailView<? extends ReferenceDto> view) {
```

Comment:
I advocate to revert the refactoring in commit 51ad5421629cb20d76ed37861d1148740f1cd92c since the details of the instantiation of a SampleListComponent are now spread across several classes (`CaseDataView`, `ContactDataView`, `EventParticipantDataView`)

Reply:
This is just an intermediary commit. Take a look at the final version of `SampleListComponent` and its usages.